src/components/RefinementList/RefinementList.tsx

```diff
@@ -63,18 +139,28 @@ class RefinementList extends Component {
       key += `/${facetValue.count}`;
     }
 
+    const refinementListItemClassName = cx(
+      this.props.cssClasses.item,
+      this.props.cssClasses.selectedItem && {
+        [this.props.cssClasses.selectedItem]: facetValue.isRefined,
+      },
+      this.props.cssClasses.disabledItem && {
+        [this.props.cssClasses.disabledItem]: !facetValue.count,
+      },
+      this.props.cssClasses.parentItem && {
+        [this.props.cssClasses.parentItem]:
+          facetValue.data && facetValue.data.length > 0,
+      }
+    );
```

Comment:
```suggestion
    const refinementListItemClassName = cx(this.props.cssClasses.item, {
      [this.props.cssClasses.selectedItem]: facetValue.isRefined,
      [this.props.cssClasses.disabledItem]: !facetValue.count,
      [this.props.cssClasses.parentItem]: hasChildren,
    });
```

this previous version seems much smaller, why did you choose this?

Reply:
Exactly. I'll see if making the component generic solves this issue, but since some classes are not always defined depending of the widget calling the component I think we will still have to do this at some point.

Reply:
considering it worked before, we can probably make them required to keep the logic simpler here (since it already works now)

Reply:
Making the component a Generic solved the issue, we can now keep the original syntax.